### ISSUE TITLE
SM MHV API isn't honoring pageSize

### ISSUE THREAD

**Issue by patrickvinograd** [department-of-veterans-affairs/va.gov-team#21454]
The MHV API supports parameters for `page` and `perPage` to control pagination, but I have a recollection that at least `perPage` was not working as expected at some point. 

Verify that we can control pagination overall. If we have to live with a fixed perPage size from MHV, that's fine, but if so we should remove that option from the vets-api endpoint so as not to introduce confusion. The most critical thing is that we can page through results in some fashion.

**Comment by patrickvinograd** [comment]
Overall, pagination is working  as expected from vets-api. 

I did notice that it seems like pagination is not working in MHV - so every request to list messages in a folder seems to retrieve all messages in a folder. vets-api then paginates those down so we're okay from the mobile app perspective. But, this could have some problems:

1) performance - fetching all messages from MHV is more expensive than just getting the subset we care about.
2) it's possible there's a lingering bug when a user has exactly 250 messages, because of this line: 
https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/sm/client.rb#L120
That line assumes that it is getting paginated results from MHV in 250-message pages. It might cause the API to go into an endless loop of retrieving the same 250 messages, or have some other weird failure mode. 

I notified the MHV team about the issues with their API - how they react will determine what we do about the above.
